quantus/metrics/base.py

```diff
@@ -101,6 +116,37 @@ def __init__(
         self.last_results: Any = []
         self.all_results: Any = []
 
+        if (
+            not hasattr(self, "_name")
+            or not hasattr(self, "_data_applicability")
+            or not hasattr(self, "_model_applicability")
+            or not hasattr(self, "_score_direction")
+            or not hasattr(self, "_evaluation_category")
+        ):
+            raise NotImplementedError(
+                "Subclasses must define name, data_applicability, model_applicability and score_direction before instantiation"
+            )
+
+    @property
```

Comment:
Why not just
```python
@property
@abstractmethod
def name(self):
     raise NotImplementedError
```
This way we won't need additional "private" properties, and will enforce inheritors to specify a name.

Reply:
Good idea, will implement.